File: SSwHost/src/RgbLedHost/SSwVBHost/Infrastructure/Power.vb
Imports System
Imports Microsoft.SPOT

Namespace SSwHost.Infrastructure
    Class Power
        Shared time As DateTime = New DateTime()

        Public Enum PeriodMax
            Second
            Minute
            FiveMinute
            FifeteenMinute
            Hour
        End Enum

        Private period As PeriodMax
        Shared current As Double
        Private ReadOnly Property defaultVoltage As Double = 120.0

        Public Sub New(ByVal time_ As DateTime, ByVal periodMax As PeriodMax, ByVal current_ As Double)
            time = time_
            period = periodMax
            current = current_
        End Sub

        Public Sub New(ByVal defaultVoltage_ As Double)
            defaultVoltage = defaultVoltage_
        End Sub
    End Class
End Namespace

File: SSwHost/Infrastructure/Power.cs
using System;
using Microsoft.SPOT;

namespace SSwHost.Infrastructure
{
    class Power
    {
        static DateTime time = new DateTime();
        public enum PeriodMax
        {
            Second,
            Minute,
            FiveMinute,
            FifeteenMinute,
            Hour
        }
        PeriodMax period;

        static double current;
        private double defaultVoltage { get; } = 120.0;
        public Power(DateTime time_, PeriodMax periodMax, double current_)
        {
            time = time_;
            period = periodMax;
            current = current_;
        }
        public Power(double defaultVoltage_)
        {
            defaultVoltage = defaultVoltage_;
        }
    }
}
